.ci/generate-releases.sh

```diff
@@ -0,0 +1,214 @@
+  jq -r --arg version "$version" '[.versions[]
+    | select(contains("SNAPSHOT")|not)
+    | select(startswith($version))]
+    | sort_by(.| split(".") | map(tonumber))
+    | .[-1]' $file
+  rm $file &> /dev/null || true
+}
+
+# Bump the patch version for the given version
+function incPatch() {
+  local version="$1"
+  version="${version#[vV]}"
+  major="${version%%\.*}"
+  minor="${version#*.}"
+  minor="${minor%.*}"
+  vpatch="${version##*.}"
+  echo "${major}.${minor}.$((vpatch + 1))"
+}
+
+# Get the latest version in the main branch.
+# It uses the https://storage.googleapis.com/artifacts-api.
+# Owned by: observability-robots
+function edge() {
+  local file=.edge
+  retry 3 curl -s https://storage.googleapis.com/artifacts-api/snapshots/main.json 2> /dev/null > $file
+  jq -r .build_id .edge | sed 's#-.*##g'
+  rm $file &> /dev/null || true
+}
+
+# Get the major version for the given semver
+function major() {
+  local version="$1"
+  version="${version#[vV]}"
+  major="${version%%.*}"
+  echo "${major}"
+}
+
+# Get the major.minor version for the given semver
+function majorminor() {
+  local version="$1"
+  version="${version#[vV]}"
+  major="${version%%.*}"
+  minor="${version#*.}"
+  minor="${minor%.*}"
+  echo "${major}.${minor}"
+}
+
+# Whether the given version is available.
+# It uses docker images in the internal docker registry
+function isAvailable() {
+  local version="$1"
+  # apm-server docker image is smaller.
+  if retry 3 docker pull --quiet docker.elastic.co/apm/apm-server:"$version"-SNAPSHOT &> /dev/null; then
+    echo 'true'
+  else
+    # Fallback to use the elasticsearch - a quite  bigger docker image.
+    if retry 3 docker pull --quiet docker.elastic.co/elasticsearch/elasticsearch:"$version"-SNAPSHOT &> /dev/null; then
+      echo 'true'
+    else
+      echo 'false'
+    fi
+  fi
+}
+
+###############
+### MAIN
+###############
+
+## 0. Static versions
+current_6="6.8.23"
+
+## 1. Fetch the versions
+current_7=$(latest v7)
+current_8=$(latest v8)
+next_7=$(incPatch "$current_7")
+next_minor_8=$(next 8)
+next_patch_8=$(incPatch "$current_8")
+edge_8=$(edge)
+
+## 2. Generate files
+
+### We avoid surprises by uploading the unexpected credentials json file
+mkdir releases
+cd releases
+
+### IMPORTANT:
+### This file might contain some versions that are not available yet.
+### One way to solve this particular case will be by reading the current releases.properties
+### and apply some validations with isAvailable, otherwise then fallback to the previous version.
+{
+  echo "current_6=$current_6"
+  echo "current_7=$current_7"
+  echo "next_minor_7=$next_7"
+  echo "next_patch_7=$next_7"
+  echo "current_8=$current_8"
+  echo "next_minor_8=$next_minor_8"
+  echo "next_patch_8=$next_patch_8"
+  echo "edge_8=$edge_8"
+  echo "generated=https://github.com/elastic/apm-pipeline-library/actions/workflows/generate-elastic-stack-releases.yml"
+} > releases.properties
```

Comment:
In a follow-up, we can do the same as done in https://github.com/elastic/apm-pipeline-library/blob/d72ac2b0f04d8de8831d52de7319f7f844a06db6/updatecli/updatecli.d/bump-aliases.yml#L92-L115

but being able to update the relevant versions when artifacts are available.